pkg/apis/rke.cattle.io/v1/rkecontrolplane_types.go

```diff
@@ -0,0 +1,180 @@
+package v1
+
+import (
+	"github.com/rancher/wrangler/v3/pkg/genericcondition"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RKEControlPlaneSpec struct {
+	ClusterConfiguration `json:",inline"`
+
+	// AgentEnvVars is a list of environment variables that will be set on
+	// the cluster agent deployment and system agent service.
+	// +optional
+	AgentEnvVars []EnvVar `json:"agentEnvVars,omitempty"`
+
+	// LocalClusterAuthEndpoint is the configuration for the local cluster
+	// auth endpoint.
+	// +optional
+	// +nullable
+	LocalClusterAuthEndpoint LocalClusterAuthEndpoint `json:"localClusterAuthEndpoint,omitempty"`
+
+	// ETCDSnapshotCreate is the configuration for the etcd snapshot creation
+	// operation.
+	// +optional
+	ETCDSnapshotCreate *ETCDSnapshotCreate `json:"etcdSnapshotCreate,omitempty"`
+
+	// ETCDSnapshotRestore is the configuration for the etcd snapshot restore
+	// operation.
+	// +optional
+	ETCDSnapshotRestore *ETCDSnapshotRestore `json:"etcdSnapshotRestore,omitempty"`
+
+	// RotateCertificates is the configuration for the certificate rotation
+	// operation.
+	// +optional
+	RotateCertificates *RotateCertificates `json:"rotateCertificates,omitempty"`
+
+	// RotateEncryptionKeys is the configuration for the encryption key
+	// rotation operation.
+	// +optional
+	RotateEncryptionKeys *RotateEncryptionKeys `json:"rotateEncryptionKeys,omitempty"`
+
+	// KubernetesVersion is the desired version of RKE2/K3s for the cluster.
+	// This field is only populated for provisioned and custom clusters.
+	// +optional
+	KubernetesVersion string `json:"kubernetesVersion,omitempty"`
+
+	// ClusterName is the name of the provisioning cluster object.
+	// +kubebuilder:validation:MaxLength=63
+	// +required
+	ClusterName string `json:"clusterName,omitempty"`
+
+	// ManagementClusterName is the name of the management cluster object
+	// that relates to this cluster.
+	// +kubebuilder:validation:Pattern=`^(c-m-[a-z0-9]{8}|local)$`
```

Comment:
See the comment on the status field for the provisioning cluster, there might be cases where this isn't true.